gptqmodel/models/base.py

```diff
@@ -236,6 +236,14 @@ def __init__(
             # setting cls.module_tree
             type(self).module_tree = apply_module_tree_override(self.module_tree, self.module_tree_overrides[quant_method])
 
+        if quant_method == METHOD.GPTQ and type(self).module_tree is None:
+            type(self).module_tree = self._auto_detect_module_tree(model)
```

Comment:
auto_detect_module_tree whould take in quant_method as param for module scanning and not have gptq hard coded in line 239.